skgstat/MetricSpace.py

```diff
@@ -243,3 +255,107 @@ def dists(self):
 
         # return
         return self._dists
+
+class ProbabalisticMetricSpace(MetricSpace):
+    """Like MetricSpace but samples the distance pairs only returning a
+       `samples` sized subset. `samples` can either be a fraction of
+       the total number of pairs (float < 1), or an integer count.
+    """
+    
+    def __init__(self, coords, dist_metric="euclidean", max_dist=None, samples=0.5):
```

Comment:
@redhog: we need another parameter here: `seed=None`. This should be passed down to any stochastic method you use (np.random.choice` I guess) to set a `np.random.seed`. That would be great to ensure reproducible samples for unittests and, more importantly, scientific applications.